ground/gcs/src/plugins/config/configautotunewidget.cpp

```diff
@@ -395,417 +333,162 @@ void ConfigAutotuneWidget::stuffShareForm(AutotuneFinalPage *autotuneShareForm)
+{
+    QWizard wizard(NULL, Qt::Dialog | Qt::CustomizeWindowHint | Qt::WindowTitleHint
+                       | Qt::WindowCloseButtonHint);
 
-    for (int i = 0; i < pts; i++) {
-        float y = b0 * (data[i] + 2.0f * x1 + x2) + a1 * y1 + a2 * y2;
+    wizard.setPixmap(QWizard::BackgroundPixmap, QPixmap(":/configgadget/images/autotunebg.png"));
 
-        y2 = y1;
-        y1 = y;
+    // Used to track state across the entire wizard path.
+    AutotunedValues av;
 
-        x2 = x1;
-        x1 = data[i];
+    if (precalc_vals) {
+        av = *precalc_vals;
+    } else {
+        memset(&av, 0, sizeof(av));
     }
 
-    for (int i = 0; i < pts; i++) {
-        float y = b0 * (data[i] + 2.0f * x1 + x2) + a1 * y1 + a2 * y2;
-
-        y2 = y1;
-        y1 = y;
+    av.converged = false;
 
-        x2 = x1;
-        x1 = data[i];
+    // The first page is simple; construct it from scratch.
+    QWizardPage *beginning = new AutotuneBeginningPage(NULL, autoOpened, &av);
 
-        data[i] = y;
-    }
-}
+    // Keep a cancel button, even on OS X.
+    wizard.setOption(QWizard::NoCancelButton, false);
 
-/* Returns number of samples of delay between series */
-float get_sample_delay(int pts, const QVector<float> &delayed, const QVector<float> &orig)
-{
-    ffft::FFTReal<float> fft(pts);
+    wizard.setMinimumSize(735, 600);
 
-    /* Convert to frequency domain */
-    QVector<float> delayed_fft(pts);
-    fft.do_fft(delayed_fft.data(), delayed.data());
+    wizard.addPage(beginning);
 
-    QVector<float> orig_fft(pts);
-    fft.do_fft(orig_fft.data(), orig.data());
+    AutotuneFinalPage *pg = new AutotuneFinalPage(NULL);
 
-    /* Now perform a correlation by multiplying -orig_fft* by delayed_fft.
-     * The types are all floats here, so we need to do the heavy lifting
-     * ourselves.   gfft = x+yi, dfft = u+vi, dfft* = u-vi,
-     * -dfft* = -u + vi
-     *
-     * -dfft* x gfft = (-ux - vy) + (vx - uy)i
-     */
+    wizard.addPage(new AutotuneMeasuredPropertiesPage(NULL, &av));
+    wizard.addPage(new AutotuneSlidersPage(NULL, &av));
 
-    QVector<float> product(pts);
+    stuffShareForm(pg);
 
-    int fpts = pts / 2;
+    wizard.addPage(pg);
 
-    // Memory layout here is annoyin'.  All reals, then all imaginaries
-    for (int i = 0; i < fpts; i++) {
-        float x = delayed_fft[i];
-        float y = delayed_fft[i + fpts];
-        float u = orig_fft[i];
-        float v = orig_fft[i + fpts];
+    wizard.setWindowTitle("Autotune Wizard");
+    wizard.exec();
 
-        product[i] = -(u * x) - (v * y);
-        product[i + fpts] = (v * x) - (u * y);
-    }
+    if (av.valid && (wizard.result() == QDialog::Accepted) && av.converged) {
+        // Apply and save data to board.
+        StabilizationSettings *stabilizationSettings =
+            StabilizationSettings::GetInstance(getObjectManager());
+        Q_ASSERT(stabilizationSettings);
+        if (!stabilizationSettings)
+            return;
```

Comment:
Can you add a qWarning here? That way we will know if there's a problem rather than it happening silently.